patch/cisco-Enable-static-memory-reservation-for-OIRable-PCIe-de.patch

```diff
@@ -0,0 +1,198 @@
++                */
++
++               /* Calculate resource based on EEPROM values */
++               ds_port_offset = (bus->number - bus->parent->number) - 1;
++               if (ds_port_offset < 5) {
++                       pci_read_config_dword(nt_virt_dev, PLX_RES_DS_PORT_REG0,
++                               &mem_res_bitmap);
++               } else {
++                       ds_port_offset -= 5;
++                       pci_read_config_dword(nt_virt_dev, PLX_RES_DS_PORT_REG1,
++                               &mem_res_bitmap);
++               }
++               port_bitmap = mem_res_bitmap;
++               dev_dbg(&bus->dev, "Port offset: 0x%x, res bitmap 0x%x\n",
++                       ds_port_offset, mem_res_bitmap);
++
++               if (ds_port_offset < 5) {
++                       u8 m[] = { 26, 20, 14, 8, 2 };
++                       u8 s[] = { 31, 25, 19, 13, 7 };
++
++                       multiplier = (port_bitmap >> m[ds_port_offset]) & PLX_RES_P_MASK;
++                       np_size = (port_bitmap >> s[ds_port_offset]) & PLX_RES_NP_MASK;
++
++                       dev_dbg(&bus->dev, "Multiplier: %d, np_size: %d\n",
++                               multiplier, np_size);
++
++                       if (res_type & IORESOURCE_PREFETCH) {
++                               size = 0x100000 << (multiplier - 1);
++                               dev_dbg(&bus->dev, "Pref Multiplier %d, Size 0x%llx\n",
++                                       multiplier, (long long) size);
++                       } else if (np_size) {
++                               size = 0x100000;
++                               dev_dbg(&bus->dev, "NP Size 0x%llx\n", (long long) size);
++                       }
++                }
++       }
++       return size;
++}
++
+ /**
+  * pbus_size_mem() - size the memory window of a given bus
+  *
+@@ -1013,6 +1156,7 @@ static int pbus_size_mem(struct pci_bus *bus, unsigned long mask,
+ 	resource_size_t children_add_size = 0;
+ 	resource_size_t children_add_align = 0;
+ 	resource_size_t add_align = 0;
++        unsigned int dev_count = 0;
+ 
+ 	if (!b_res)
+ 		return -ENOSPC;
+@@ -1024,6 +1168,7 @@ static int pbus_size_mem(struct pci_bus *bus, unsigned long mask,
+ 	list_for_each_entry(dev, &bus->devices, bus_list) {
+ 		int i;
+ 
++                dev_count++;
+ 		for (i = 0; i < PCI_NUM_RESOURCES; i++) {
+ 			struct resource *r = &dev->resource[i];
+ 			resource_size_t r_size;
+@@ -1077,6 +1222,21 @@ static int pbus_size_mem(struct pci_bus *bus, unsigned long mask,
+ 		}
+ 	}
+ 
++        /* Static allocation for FC pacific */
++        if (!size && !dev_count) {
++            size = pci_get_plx_downstream_res_size(bus, type);
++            if (size) {
++                order = __ffs(size);
++                dev_dbg(&bus->self->dev, "order for %llx is %u\n", (long long) size, order);
++                if ((order >= 20) &&
++                    ((order -= 20) < ARRAY_SIZE(aligns)) &&
```

Comment:
`checkpatch.pl` complains about the `order -= 20`:

    ERROR: do not use assignment in if condition
    #203: FILE: drivers/pci/setup-bus.c:1224:
    +			if ((order >= 20) &&

Reply:
Fixed coding style issues and verified running checkpatch.pl